File: study/ts/script/ui/component/CommonJadeProperty.ts
const { ccclass, property } = cc._decorator;

import CommonDetailWindow from './CommonDetailWindow'
import JadeAttrNode from '../../scene/view/equipmentJade/JadeAttrNode';
import JadeSuitEquipmentNode from '../../scene/view/equipmentJade/JadeSuitEquipmentNode';
import { TypeConvertHelper } from '../../utils/TypeConvertHelper';

@ccclass
export default class CommonJadeProperty extends cc.Component {

    @property({
        type: CommonDetailWindow,
        visible: true
    })
    _commonDetailWindow: CommonDetailWindow = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _name2: cc.Label = null;

    @property({
        type: JadeSuitEquipmentNode,
        visible: true
    })
    jadeSuitEquipmentNode: JadeSuitEquipmentNode = null;

    @property({
        type: JadeAttrNode,
        visible: true
    })
    jadeAttrNode: JadeAttrNode = null;

    _listViewHeight: number;

    _listView: cc.Node;
    _jadeConfig: any;

    onLoad() {
        this._listView = this._commonDetailWindow._listView;
    }

    _buildAttrModule() {
        this.jadeAttrNode.setJadeData(this._jadeConfig);
        this.jadeAttrNode.node.setPosition(0, -this._listViewHeight);
        this._listViewHeight += this.jadeAttrNode._listView.height;
    }
    _buildSuitEquipments() {
        this.jadeSuitEquipmentNode.setJadeData(this._jadeConfig);
        this.jadeSuitEquipmentNode.node.setPosition(0, -this._listViewHeight);
        this._listViewHeight += this.jadeSuitEquipmentNode._listView.height;
    }
    _updateListView() {
        this._buildAttrModule();
        this._buildSuitEquipments();
    }
    updateUI(jadeConfig) {
        this._listViewHeight = 0;
        this._jadeConfig = jadeConfig;
        this._updateListView();
        this._listView.height = this._listViewHeight;
        var param = TypeConvertHelper.convert(TypeConvertHelper.TYPE_JADE_STONE, jadeConfig.id);
        this._name2.string = (param.name);
        this._name2.node.color = (param.icon_color);
    }

}

File: study/lua/src/app/ui/component/CommonJadeProperty.lua
local CommonJadeProperty = class("CommonJadeProperty")
local JadeAttrNode = require("app.scene.view.equipmentJade.JadeAttrNode")
local JadeSuitEquipmentNode = require("app.scene.view.equipmentJade.JadeSuitEquipmentNode")

local EXPORTED_METHODS = {
    "updateUI"
}

function CommonJadeProperty:ctor()
    self._target = nil
    self._name2 = nil
end

function CommonJadeProperty:_init()
    self._name2 = ccui.Helper:seekNodeByName(self._target, "Name2")
    self._listView = ccui.Helper:seekNodeByName(self._target, "ListView")
    self._listView:setScrollBarEnabled(false)
    cc.bind(self._target, "CommonDetailWindow")
end

function CommonJadeProperty:bind(target)
    self._target = target
    self:_init()
    cc.setmethods(target, self, EXPORTED_METHODS)
end

function CommonJadeProperty:unbind(target)
    cc.unsetmethods(target, EXPORTED_METHODS)
end

--玉石属性
function CommonJadeProperty:_buildAttrModule()
    local jadeConfig = self._jadeConfig
    local jadeAttrNode = JadeAttrNode.new(jadeConfig)
    self._listView:pushBackCustomItem(jadeAttrNode)
end

function CommonJadeProperty:_buildSuitEquipments()
    local jadeConfig = self._jadeConfig
    local jadeSuitEquipmentNode = JadeSuitEquipmentNode.new(jadeConfig)
    self._listView:pushBackCustomItem(jadeSuitEquipmentNode)
end

function CommonJadeProperty:_updateListView()
    --详情List开始
    self._listView:removeAllChildren()
    --属性
    self:_buildAttrModule()
    --适用装备
    self:_buildSuitEquipments()
end

function CommonJadeProperty:updateUI(jadeConfig)
    self._jadeConfig = jadeConfig
    self:_updateListView()
    local TypeConvertHelper = require("app.utils.TypeConvertHelper")
    local param = TypeConvertHelper.convert(TypeConvertHelper.TYPE_JADE_STONE, jadeConfig.id)
    self._name2:setString(param.name)
    self._name2:setColor(param.icon_color)
    -- self._name2:enableOutline(param.icon_color_outline, 2)
end

return CommonJadeProperty
